PyAssembler/Assembler.py

```diff
@@ -0,0 +1,421 @@
+        """ 
+        Removes the comments (if exists) in each assembly code line.
+        
+        :param asm_lines: A list of extracted assembly lines.
+        :returns: A comments-cleaned version of the assembly lines
+        :rtype: List of strings
+        """
+        for index, line in enumerate(asm_lines):
+            line = line.strip()
+            
+            # Blank lines or Comment lines
+            if line == '' or line.startswith(Assembler.COMMENT_PREFIX):
+                asm_lines[index] = None
+            else:
+                # Inline comments
+                code, *comment = line.split(Assembler.COMMENT_PREFIX, maxsplit=1)
+                asm_lines[index] = code.strip()
+        
+        return [line for line in asm_lines if line is not None]
+
+    def first_step(self, assembly_lines):
+        """ 
+        Performs the first step in the assembling process. It consist of translation
+        of the code lines (except for the usages of labels), and storage of labels and their addresses.
+        
+        :param assembly_lines: A list of assembly code lines strings.
+        :returns: The partially-translated lines.
+        :rtype: list of bytearrays
+        """
+        cleaned_lines = self.clean_comments(assembly_lines)
+        semi_trans_lines = [self.translate_line(line) for line in cleaned_lines]
+        return semi_trans_lines
+
+    def second_step(self, semi_trans_lines):
+        """ 
+        Performs the second step in the assembling process. It resolves the correct
+        address value of a label and evaluates expressions of which it contained in them.
+        
+        :param semi_trans_lines: A list of partially-translated lines.
+        :returns: The completely translated lines
+        :rtype: list of bytearrays
+        """
+        for line_num in self.untranslated_exps:
+            eval_expr = eval(self.untranslated_exps[line_num], self.sym_tbl)
+            semi_trans_lines[line_num].extend(eval_expr.to_bytes(Assembler.NUM_BYTES, byteorder=Assembler.BIG_ENDIAN))
+        return semi_trans_lines
+
+    def assemble_code(self, assembly_lines):
+         """ 
+        Performes the whole assembling process.
+        
+        :param assembly_lines: A list of the extracted assembly lines, as strings.
+        :returns: The completely translated lines to binary bytes.
+        :rtype: list of bytearrays
+        """
+        half_translated_lines = self.first_step(assembly_lines)
+        return self.second_step(half_translated_lines)
+
+
+
+def setup_argparse():
+    parser = argparse.ArgumentParser()
+    
+    parser.add_argument("-i", "--input", help="Directs the input to a name of your choice", required=True)
+    parser.add_argument("-o", "--output", help="Directs the output to a name of your choice", required=True)
+    
+    return parser.parse_args()
+
+if __name__ == '__main__':
+    args = setup_argparse()
+    asm = Assembler()
+    all_lines = []

```

Comment:
Unnecessary, Python doesn't need you to predefine variables. Most Python statements don't have a "scope" (only exclusions are `def` and `class`), so it's OK to just declare variables inside `with` statements (this is also the common practice).